24/09/25 changes

mamurak opened the pull request:
Mainly commits by @mbaldessari. Added fixes to the dev project manifest.

mbaldessari committed:
Drop the operator-install folder (pattern-install chart)

mbaldessari committed:
Start using the OCI chart in lieu of operator-install

mbaldessari committed:
Deploy with gitea by default

mbaldessari committed:
Add a README note on the main branch

mbaldessari committed:
Drop the chart splitting workflows

We do not push any changes from common to the chart because they do not
exist in common any longer

mbaldessari committed:
Drop the test folder

It used to contain expected tests for the different charts, but those
have moved away, so let's start dropping these for now.

mbaldessari committed:
Drop the json schema checking job

It only makes sense in the presence of helm charts. We should prolly
move it to each chart repo.

mbaldessari committed:
Merge pull request #591 from mbaldessari/main-update-readme

Add a README note on the main branch

mbaldessari committed:
Merge pull request #592 from mbaldessari/slimming

Slim down common

mbaldessari committed:
Merge remote-tracking branch 'common-upstream/main' into common-automatic-update

mbaldessari committed:
Merge pull request #273 from mbaldessari/common-automatic-update

Automatic common/ update from branch main

mbaldessari committed:
Add a check to see if multisource for clustergroup is enabled

Without this in your values-global.yaml files, the deployment with a
slimmed down common would fail as follows:

    - lastTransitionTime: "2024-09-13T18:30:19Z"
    message: 'Failed to load target state: failed to generate manifest for source
      1 of 1: rpc error: code = Unknown desc = Manifest generation error (cached):
      common/clustergroup: app path does not exist'

mbaldessari committed:
Merge pull request #593 from mbaldessari/check-for-multisource-enabled

Add a check to see if multisource for clustergroup is enabled

mbaldessari committed:
Move the common slim + multisource test into Makefile

This way yq is not required on the host. Tested as follows:

* No value set (assumes default is false)

❯ cat values-global.yaml
---
global:
  pattern: multicloud-gitops
  options:
    useCSV: false
    syncPolicy: Automatic
    installPlanApproval: Automatic
main:
  clusterGroupName: hub
  #  multiSourceConfig:
  #  enabled: true

❯ ./pattern.sh make validate-prereq
make -f common/Makefile validate-prereq
make[1]: Entering directory '/home/michele/Engineering/cloud-patterns/multicloud-gitops'
You must set ".main.multiSourceConfig.enabled: true" in your 'values-global.yaml' file
because your common subfolder is the slimmed down version with no helm charts in it
make[1]: *** [common/Makefile:161: validate-prereq] Error 1
make[1]: Leaving directory '/home/michele/Engineering/cloud-patterns/multicloud-gitops'
make: *** [Makefile:12: validate-prereq] Error 2

* Value set to false

❯ cat values-global.yaml
---
global:
  pattern: multicloud-gitops
  options:
    useCSV: false
    syncPolicy: Automatic
    installPlanApproval: Automatic
main:
  clusterGroupName: hub
  multiSourceConfig:
    enabled: false
❯ ./pattern.sh make validate-prereq
make -f common/Makefile validate-prereq
make[1]: Entering directory '/home/michele/Engineering/cloud-patterns/multicloud-gitops'
You must set ".main.multiSourceConfig.enabled: true" in your 'values-global.yaml' file
because your common subfolder is the slimmed down version with no helm charts in it
make[1]: *** [common/Makefile:161: validate-prereq] Error 1
make[1]: Leaving directory '/home/michele/Engineering/cloud-patterns/multicloud-gitops'
make: *** [Makefile:12: validate-prereq] Error 2

* Value set to true

❯ cat values-global.yaml
---
global:
  pattern: multicloud-gitops
  options:
    useCSV: false
    syncPolicy: Automatic
    installPlanApproval: Automatic
main:
  clusterGroupName: hub
  multiSourceConfig:
    enabled: true
❯ ./pattern.sh make validate-prereq
make -f common/Makefile validate-prereq
make[1]: Entering directory '/home/michele/Engineering/cloud-patterns/multicloud-gitops'
make[1]: Leaving directory '/home/michele/Engineering/cloud-patterns/multicloud-gitops'

mbaldessari committed:
Merge pull request #594 from mbaldessari/fix

Move the common slim + multisource test into Makefile

mbaldessari committed:
Drop last bit of operator-install and the common symlink

mbaldessari committed:
Merge pull request #595 from mbaldessari/drop-link

Drop last bit of operator-install and the common symlink

mbaldessari committed:
Drop reference-output.yaml as it makes little sense in the slimmed common

mbaldessari committed:
Merge pull request #596 from mbaldessari/drop-reference

Drop reference-output.yaml as it makes little sense in the slimmed common

mbaldessari committed:
Merge remote-tracking branch 'common-upstream/main' into common-automatic-update

mbaldessari committed:
Merge pull request #274 from mbaldessari/common-automatic-update

Automatic common/ update from branch main

mbaldessari committed:
Merge upstream

mbaldessari committed:
Use gitea by default in my env

mbaldessari committed:
Only use the pushsecret on the hub and when gitea is being used

Commit:
Remove ansible code in favor of external collection

mbaldessari committed:
Also have the gitea admin secret in the ml-development ns

Commit:
Make adjustments for ansible code moving into separate collection repo

mbaldessari committed:
pin odf source

mbaldessari committed:
Switch to our ODF chart

mamurak committed:
Merge branch 'integrate_rhoai-wip' into mamurak/integrate_rhoai_wip

mbaldessari committed:
any nodes

mbaldessari committed:
any nodes

mamurak committed:
added gitea admin secret to workbench

mbaldessari committed:
any nodes

mbaldessari committed:
any nodes 2

mbaldessari committed:
Merge branch 'integrate_rhoai-wip' of https://github.com/mhjacks/industrial-edge into martin-integrate_rhoai-wip

mbaldessari committed:
ops

mbaldessari committed:
Add my baremetal overrides

mbaldessari committed:
test

mbaldessari committed:
test2

mbaldessari committed:
Merge pull request #597 from mhjacks/factor_out_ansible

Factor out ansible into separate collection repo

mbaldessari committed:
Add a note about the changes in common

mbaldessari committed:
Merge pull request #600 from mbaldessari/add-changes

Add a note about the changes in common

mamurak committed:
Merge remote-tracking branch 'mbaldessari/martin-integrate_rhoai-wip' into mamurak/integrate_rhoai_wip

mamurak committed:
deduplicated envfrom in devproject

mbaldessari committed:
Add qe-tests target

beekhof committed:
Remove assorted detritus

beekhof committed:
We don't have charts to test anymore

mbaldessari committed:
Merge pull request #601 from validatedpatterns/cleanup

Remove assorted detritus

mbaldessari committed:
Drop lint.sh and its target in Makefile

mbaldessari committed:
Drop test.sh and its target in Makefile and also kubeconform

mbaldessari committed:
Merge pull request #603 from mbaldessari/more-cleanups

More cleanups

mbaldessari committed:
Merge remote-tracking branch 'remote-common/main' into slimming

mbaldessari committed:
Use the new clusterGroupChart 0.9.* version

mbaldessari committed:
Drop obsolete Makefile target

mbaldessari committed:
Merge branch 'main' into qe-test

mbaldessari committed:
Merge pull request #275 from mbaldessari/slimming

Switchover to slimmed common

mbaldessari committed:
Merge pull request #602 from mbaldessari/qe-test

Add qe-tests target

mbaldessari commented:
Ah best if we don't merge from my rhoi wip branch, I was mainly testing some local stuff. I'll push proper PRs your way if it is mergeable stuff (sorry if I created any confusion)

mbaldessari committed:
Read PATTERN_INSTALL_CHART from the env as well if it exists

mbaldessari committed:
Merge pull request #604 from mbaldessari/disconnected-inst

Read PATTERN_INSTALL_CHART from the env as well if it exists

mbaldessari committed:
Revert "test2"

This reverts commit c13244a7cf96b5e23d93be151c5550e656c9fba3.

mbaldessari committed:
Revert "test"

This reverts commit 23dd9a3db2e64e9fe0e749dcdc021e4f9247096d.

mbaldessari committed:
Revert "Add my baremetal overrides"

This reverts commit b69a9fae730478bb22e532f38ace97caf31125cf.

mbaldessari committed:
Revert "ops"

This reverts commit e36fe95217f1a1ee4cfea1def6226567db326854.

mbaldessari committed:
Revert "any nodes 2"

This reverts commit 2fc22293ada6911c60011fd9a129a1c9399e9d8d.

mbaldessari committed:
Revert "any nodes"

This reverts commit 4efd8272b1552167c1fa79e65cac7b55353a829f.

mbaldessari committed:
Revert "any nodes"

This reverts commit 189664a11b325729478b56c7fca2791285182f31.

mbaldessari committed:
Revert "any nodes"

This reverts commit aa23dfb10a2dceb4b171324b98180ab5017602dc.

mbaldessari committed:
Revert "Switch to our ODF chart"

This reverts commit a9e129abb95a3d041c8ef09f8fbf08b46da3621a.

darkdoc committed:
Update secret backend change script

The script is used in make secrets-backend-*. Update needed for slimed
common.

darkdoc committed:
Update preview script

Now the preview will work with remote charts, and local charts as well.
Default helm registry will be oci://quay.io/repository/hybridcloudpatterns/.

mbaldessari committed:
Merge pull request #605 from darkdoc/fix_scripts

Fix scripts after common slim

mbaldessari committed:
Introduce PATTERN_DISCONNECTED_HOME

When doing disconnected installations this allows for setting a single
variable pointing to the internal disconnected registry.

mbaldessari committed:
Merge pull request #606 from mbaldessari/disconnected-easier

Introduce PATTERN_DISCONNECTED_HOME

soukron committed:
Update Makefile to reflect a warning if no storageclass is present

mbaldessari committed:
Merge pull request #608 from soukron/feature/disable-storage-class-requirement

Update Makefile to reflect a warning if no storageclass is present

mbaldessari committed:
Try and add proper custom endpoint support

Co-Authored-By: Akos Eros <aeros@redhat.com>

mamurak committed:
Merge pull request #2 from mbaldessari/integrate_rhoai_wip_max

Deploy with gitea by default

mamurak committed:
Merge pull request #3 from mbaldessari/for-max

Some reverts of commits that were not ready yet

mamurak committed:
added modelmesh serving runtimes

mamurak committed:
removed broken serving manifests

mamurak commented:
added commits by @mbaldessari and serving runtime modifications

mbaldessari committed:
Fix iib loading playbook name

After slimming common down the playbook needs renaming because
collections have slightly different naming constraints. Previous error:

❯ ./pattern.sh make load-iib
make -f common/Makefile load-iib
make[1]: Entering directory '/home/michele/Engineering/cloud-patterns/multicloud-gitops'
ERROR! the playbook: rhvp.cluster_utils.iib-ci could not be found
make[1]: *** [common/Makefile:117: load-iib] Error 1
make[1]: Leaving directory '/home/michele/Engineering/cloud-patterns/multicloud-gitops'
make: *** [Makefile:12: load-iib] Error 2

mbaldessari committed:
Merge pull request #609 from mbaldessari/fix-iib

Fix iib loading playbook name

mbaldessari committed:
Merge remote-tracking branch 'common-upstream/main' into common-automatic-update

mbaldessari committed:
Merge pull request #276 from mbaldessari/common-automatic-update

common automatic update

mbaldessari committed:
Switch to community operator for camel-k

When writing to the s3 endpoint exported by nooba, we need a feature in
camel called `forcePathStyle`, because the s3 endpoint does not work
with virtual hosts like proper S3 from AWS.

This `forcePathStyle` support showed up in camel-3.21.x. Currently
the downstream camel-k only ships version 3.18.6 which will error out
as follows:

    Failed to resolve endpoint:
    aws2-s3://anomaly-detection?accessKey=xxxxxx&forcePathStyle=true&overrideEndpoint=true&region=none&secretKey=xxxxxx&uriEndpointOverride=http%3A%2F%2Fs3.openshift-storage.svc
    due to: There are 1 parameters that couldn't be set on the endpoint.
    Check the uri if the parameters are spelt correctly and that they are
    properties of the endpoint. Unknown parameters=[{forcePathStyle=true}]

There are no newer runtimes we can use in the CamelCatalog provided by
the camel-k downstream operator currently (operator version
1.10.7+xxxx).

So for the time being let's switch to the upstream camel k operator
which ships a camel catalog that is more recent:

    oc get -n openshift-operators camelcatalogs camel-catalog-3.8.1 -o yaml |grep camel.version
    camel.version: 4.4.1

With this version we're able to set the proper S3 URI and the
application is able to write to the nooba s3 bucket correctly.

Co-Authored-By: Akos Eros <aeros@redhat.com>

mbaldessari committed:
Add bucketname

This way the bucketname is stable and no UUID get appended

Co-Authored-By: Akos Eros <aeros@redhat.com>

mbaldessari committed:
Drop the global s3 bucket bits

Make them normal chart parameters for the time being

mbaldessari committed:
Drop s3-secret

This was needed when we pushed things to AWS S3.
Now that we push to the ObjectBucketClaim called anomaly-detection,
this is not needed anymore.

mbaldessari committed:
Switch to the external ODF chart

Tested and everything seems to be fine

mbaldessari committed:
Drop old MirrorMaker in favour of MirrorMaker2 and fix the policy for the kafka on hub CA

Drop the old mirrormaker and use the newer mirrormaker 2

We also fix the name of the kafka endpoint on the hub and set it to
prod-kafka-cluster-kafka-bootstrap-manuela-data-lake.

With these changes we could see the factory sending sensor readings to
the main kafka and then end up on the S3 bucket

Co-Authored-By: Akos Eros <aeros@redhat.com>

mbaldessari committed:
Drop unused tests

After switching to the slimmed version of common
we can drop these tests as they are unused anyways.

mbaldessari committed:
Merge pull request #277 from mbaldessari/drop-unused-tests

Drop unused tests

mbaldessari committed:
Merge remote-tracking branch 'upstream/main' into for-max-all-fixes

mbaldessari committed:
Drop seldon subscription from the factory

It does not exist anyways (on ocp > 4.13), so no point in it being there

mbaldessari committed:
Drop mistaken clusterGroupChartVersion in the wrong section

mbaldessari committed:
Drop external-secrets argo application

Move the remaining two templates where they belong which is the
pipelines application

mbaldessari committed:
Drop unused tests

mbaldessari committed:
Drop objectBucketName

It is confusing argo and this puts it in line with the other two
objectbucketclaims

mbaldessari committed:
Drop markdown prettier from superlinter

mbaldessari committed:
Improve validate-origin

If we set the main.git.upstreamRepoURL then we need to check that one
for the presence of the branch and not whatever is configured as the
origin remote.

Tested as follows:

- Working upstreamRepoURL:

    ❯ yq -r '.main.git.repoUpstreamURL // (.main.git.repoUpstreamURL = "")' values-global.yaml
    https://github.com/mbaldessari/industrial-edge
    ❯ make validate-origin
    make -f common/Makefile validate-origin
    make[1]: Entering directory '/home/michele/Engineering/cloud-patterns/industrial-edge'
    Checking repository:
    Upstream URL set to: https://github.com/mbaldessari/industrial-edge
      https://github.com/mbaldessari/industrial-edge - branch 'for-max-s3-kafka-fix': OK
    make[1]: Leaving directory '/home/michele/Engineering/cloud-patterns/industrial-edge'

- upstreamRepoURL without the branch:

    ❯ yq -r '.main.git.repoUpstreamURL // (.main.git.repoUpstreamURL = "")' values-global.yaml
    https://github.com/mamurak/industrial-edge
    ❯ make validate-origin
    make -f common/Makefile validate-origin
    make[1]: Entering directory '/home/michele/Engineering/cloud-patterns/industrial-edge'
    Checking repository:
    Upstream URL set to: https://github.com/mamurak/industrial-edge
      https://github.com/mamurak/industrial-edge - branch 'for-max-s3-kafka-fix': NOT FOUND
    make[1]: *** [common/Makefile:138: validate-origin] Error 1
    make[1]: Leaving directory '/home/michele/Engineering/cloud-patterns/industrial-edge'
    make: *** [Makefile:24: validate-origin] Error 2

- No repoUpstreamURL set:

    ❯ yq -r '.main.git.repoUpstreamURL // (.main.git.repoUpstreamURL = "")' values-global.yaml
    ❯
    ❯ make validate-origin
    make -f common/Makefile validate-origin
    make[1]: Entering directory '/home/michele/Engineering/cloud-patterns/industrial-edge'
    Checking repository:
      https://github.com/mbaldessari/industrial-edge.git - branch 'for-max-s3-kafka-fix': OK
    make[1]: Leaving directory '/home/michele/Engineering/cloud-patterns/industrial-edge'

mbaldessari committed:
Improve validate-origin

If we set the main.git.upstreamRepoURL then we need to check that one
for the presence of the branch and not whatever is configured as the
origin remote.

Tested as follows:

- Working upstreamRepoURL:

    ❯ yq -r '.main.git.repoUpstreamURL // (.main.git.repoUpstreamURL = "")' values-global.yaml
    https://github.com/mbaldessari/industrial-edge
    ❯ make validate-origin
    make -f common/Makefile validate-origin
    make[1]: Entering directory '/home/michele/Engineering/cloud-patterns/industrial-edge'
    Checking repository:
    Upstream URL set to: https://github.com/mbaldessari/industrial-edge
      https://github.com/mbaldessari/industrial-edge - branch 'for-max-s3-kafka-fix': OK
    make[1]: Leaving directory '/home/michele/Engineering/cloud-patterns/industrial-edge'

- upstreamRepoURL without the branch:

    ❯ yq -r '.main.git.repoUpstreamURL // (.main.git.repoUpstreamURL = "")' values-global.yaml
    https://github.com/mamurak/industrial-edge
    ❯ make validate-origin
    make -f common/Makefile validate-origin
    make[1]: Entering directory '/home/michele/Engineering/cloud-patterns/industrial-edge'
    Checking repository:
    Upstream URL set to: https://github.com/mamurak/industrial-edge
      https://github.com/mamurak/industrial-edge - branch 'for-max-s3-kafka-fix': NOT FOUND
    make[1]: *** [common/Makefile:138: validate-origin] Error 1
    make[1]: Leaving directory '/home/michele/Engineering/cloud-patterns/industrial-edge'
    make: *** [Makefile:24: validate-origin] Error 2

- No repoUpstreamURL set:

    ❯ yq -r '.main.git.repoUpstreamURL // (.main.git.repoUpstreamURL = "")' values-global.yaml
    ❯
    ❯ make validate-origin
    make -f common/Makefile validate-origin
    make[1]: Entering directory '/home/michele/Engineering/cloud-patterns/industrial-edge'
    Checking repository:
      https://github.com/mbaldessari/industrial-edge.git - branch 'for-max-s3-kafka-fix': OK
    make[1]: Leaving directory '/home/michele/Engineering/cloud-patterns/industrial-edge'

mbaldessari committed:
Drop OBC job

The object bucket claim job creates secrets starting from the
noobaa generated secret which contains only key_id + secret and also add
the three AWS_S3_BUCKET, AWS_S3_ENDPOINT, AWS_DEFAULT_REGION fields.

The code lives here https://github.com/mamurak/os-mlops/blob/main/container-images/obc-dataconnection-init/obc_dataconnection_init.py

Let's switch to ESO pushsecrets + externalsecret, this way:
1) We do not rely on jobs
2) We are completely idempotent
3) We rely on less components

mbaldessari committed:
Rename secrets

This way things are a bit simpler. Also drop unused value

mbaldessari committed:
Move repoUpstreamURL to the upstream org repo

mbaldessari committed:
Add a readme entry about this branch

mbaldessari committed:
Add labels

mbaldessari committed:
Merge pull request #278 from mbaldessari/new-pr

Add proper labels

mlabonte-rh committed:
Updates for upstream testing

mlabonte-rh committed:
Updates for upstream testing

mbaldessari committed:
Make the make install output less daunting

The current output is a bit daunting for first-time users as it outputs
things like the following a few times:

    customresourcedefinition.apiextensions.k8s.io/patterns.gitops.hybrid-cloud-patterns.io created
    configmap/patterns-operator-config created
    subscription.operators.coreos.com/patterns-operator created
    error: resource mapping not found for name: "rhoai-patterns-demo" namespace: "openshift-operators" from "STDIN": no matches for kind "Pattern" in version "gitops.hybrid-cloud-patterns.io/v1alpha1"
    ensure CRDs are installed first

Let's switch to something a bit more user-friendly:

    make -f common/Makefile operator-deploy
    make[1]: Entering directory '/home/michele/Engineering/cloud-patterns/multicloud-gitops'
    Checking repository:
      https://github.com/mbaldessari/multicloud-gitops.git - branch 'luis-demo': OK
    Checking cluster:
      cluster-info: OK
      storageclass: WARNING: No storageclass found
    Installing pattern: Done
    ...

Do some magic with file descriptors so we still manage to capture the
helm template stderr and the oc apply stdout+stderr and output them at
the end in case of failure.

In such cases the output will be something like the following:

    Installing pattern: .....Installation failed [5/5]. Error:
    Pulled: quay.io/hybridcloudpatterns/pattern-install:0.0.3
    Digest: sha256:dd2d35d462b75aa8358ff278757dca0ee3c878cadafa64df8c68f880b59569ef
    E1015 18:41:31.585465  196315 memcache.go:265] couldn't get current server API group list: Get "https://api.sno3.ocplab.ocp:6443/api?timeout=32s": tls: failed to verify certificate: x509: certificate signed by un
    known authority (possibly because of "crypto/rsa: verification error" while trying to verify candidate authority certificate "kube-apiserver-lb-signer")
    ...

Suggested-by: Luis Tomas Bolivar <ltomasbo@redhat.com>

mbaldessari committed:
Make the make install output less daunting

The current output is a bit daunting for first-time users as it outputs
things like the following a few times:

    customresourcedefinition.apiextensions.k8s.io/patterns.gitops.hybrid-cloud-patterns.io created
    configmap/patterns-operator-config created
    subscription.operators.coreos.com/patterns-operator created
    error: resource mapping not found for name: "rhoai-patterns-demo" namespace: "openshift-operators" from "STDIN": no matches for kind "Pattern" in version "gitops.hybrid-cloud-patterns.io/v1alpha1"
    ensure CRDs are installed first

Let's switch to something a bit more user-friendly:

    make -f common/Makefile operator-deploy
    make[1]: Entering directory '/home/michele/Engineering/cloud-patterns/multicloud-gitops'
    Checking repository:
      https://github.com/mbaldessari/multicloud-gitops.git - branch 'luis-demo': OK
    Checking cluster:
      cluster-info: OK
      storageclass: WARNING: No storageclass found
    Installing pattern: Done
    ...

Do some magic with file descriptors so we still manage to capture the
helm template stderr and the oc apply stdout+stderr and output them at
the end in case of failure.

In such cases the output will be something like the following:

    Installing pattern: .....Installation failed [5/5]. Error:
    Pulled: quay.io/hybridcloudpatterns/pattern-install:0.0.3
    Digest: sha256:dd2d35d462b75aa8358ff278757dca0ee3c878cadafa64df8c68f880b59569ef
    E1015 18:41:31.585465  196315 memcache.go:265] couldn't get current server API group list: Get "https://api.sno3.ocplab.ocp:6443/api?timeout=32s": tls: failed to verify certificate: x509: certificate signed by un
    known authority (possibly because of "crypto/rsa: verification error" while trying to verify candidate authority certificate "kube-apiserver-lb-signer")
    ...

Suggested-by: Luis Tomas Bolivar <ltomasbo@redhat.com>

mbaldessari committed:
Merge pull request #279 from validatedpatterns/upstream_ci_fixes

Upstream ci fixes

mbaldessari committed:
Merge pull request #610 from mbaldessari/bettercheck

Improve validate-origin

mbaldessari committed:
Merge pull request #611 from mbaldessari/silence-warning

Make the make install output less daunting

mbaldessari committed:
Record the exit code at the right time

Otherwise it records the exit code of the exec call which is always 0.

This now makes things error out correctly whereas previously it did not:

    ❯ ./pattern.sh make install
    make -f common/Makefile operator-deploy
    make[1]: Entering directory '/home/michele/Engineering/cloud-patterns/multicloud-gitops'
    Checking repository:
      https://github.com/mbaldessari/multicloud-gitops.git - branch 'debug': OK
    Checking cluster:
      cluster-info: OK
      storageclass: OK
    Installing pattern: .....Installation failed [5/5]. Error:
    Pulled: quay.io/rhn_support_mbaldess/pattern-install:0.0.4
    Digest: sha256:6a5431d257f4bf05d6ac91bca4f4aa010ecb1ee60d2e03c1d0984835d210a0a8

    customresourcedefinition.apiextensions.k8s.io/patterns.gitops.hybrid-cloud-patterns.io unchanged
    configmap/patterns-operator-config unchanged
    resource mapping not found for name: "patterns-operator" namespace: "openshift-operators" from "STDIN": no matches for kind "SubscriptionNotexistant" in version "operators.coreos.com/v1alpha1"
    ensure CRDs are installed first
    Error from server (NotFound): error when creating "STDIN": namespaces "openshift-operators-notexistant" not found
    make[1]: *** [common/Makefile:71: operator-deploy] Error 1
    make[1]: Leaving directory '/home/michele/Engineering/cloud-patterns/multicloud-gitops'
    make: *** [Makefile:12: operator-deploy] Error 2

mbaldessari committed:
Record the exit code at the right time

Otherwise it records the exit code of the exec call which is always 0.

This now makes things error out correctly whereas previously it did not:

    ❯ ./pattern.sh make install
    make -f common/Makefile operator-deploy
    make[1]: Entering directory '/home/michele/Engineering/cloud-patterns/multicloud-gitops'
    Checking repository:
      https://github.com/mbaldessari/multicloud-gitops.git - branch 'debug': OK
    Checking cluster:
      cluster-info: OK
      storageclass: OK
    Installing pattern: .....Installation failed [5/5]. Error:
    Pulled: quay.io/rhn_support_mbaldess/pattern-install:0.0.4
    Digest: sha256:6a5431d257f4bf05d6ac91bca4f4aa010ecb1ee60d2e03c1d0984835d210a0a8

    customresourcedefinition.apiextensions.k8s.io/patterns.gitops.hybrid-cloud-patterns.io unchanged
    configmap/patterns-operator-config unchanged
    resource mapping not found for name: "patterns-operator" namespace: "openshift-operators" from "STDIN": no matches for kind "SubscriptionNotexistant" in version "operators.coreos.com/v1alpha1"
    ensure CRDs are installed first
    Error from server (NotFound): error when creating "STDIN": namespaces "openshift-operators-notexistant" not found
    make[1]: *** [common/Makefile:71: operator-deploy] Error 1
    make[1]: Leaving directory '/home/michele/Engineering/cloud-patterns/multicloud-gitops'
    make: *** [Makefile:12: operator-deploy] Error 2

darkdoc committed:
Merge pull request #612 from mbaldessari/fix-issue

Record the exit code at the right time

mbaldessari committed:
Switch to ACM 2.11

mbaldessari committed:
Merge pull request #280 from mbaldessari/update-acm

Switch to ACM 2.11

mamurak committed:
overriding upstream repo reference

mbaldessari committed:
Add the git branch to the gitea_admin secret

This is needed by the notebooks to push back to gitea on the correct
branch

mbaldessari committed:
Add the git branch to the gitea_admin secret

This is needed by the notebooks to push back to gitea on the correct
branch

mbaldessari committed:
Merge pull request #281 from mbaldessari/wip-next-gen-bandini2

Add branch to gitea-admin secret

mamurak committed:
added serving runtime and route of inference service in ml-development

mlabonte-rh committed:
Fix for renamed API lib

mlabonte-rh committed:
Merge pull request #282 from validatedpatterns/update_lib

Fix for renamed API lib

mbaldessari committed:
Drop double manuela-ci namespace

This kills the warning on the clusterwide argo:

    Namespace/manuela-ci is part of applications openshift-gitops/industrial-edge-datacenter and manuela

mbaldessari committed:
Drop double manuela-ci namespace

This kills the warning on the clusterwide argo:

    Namespace/manuela-ci is part of applications openshift-gitops/industrial-edge-datacenter and manuela

mbaldessari committed:
Merge pull request #283 from mbaldessari/wip-next-gen-bandini2

Drop double manuela-ci namespace

mbaldessari committed:
Revert "Drop double manuela-ci namespace"

This reverts commit 2568e0ccd83ac02280e7356f748051f4545b5b5c.

manuela-ci namespace is needed by the pipeline-setup target. Until we
figure that out properly let's revert this.

mbaldessari committed:
Revert "Drop double manuela-ci namespace"

This reverts commit 2568e0ccd83ac02280e7356f748051f4545b5b5c.

manuela-ci namespace is needed by the pipeline-setup target. Until we
figure that out properly let's revert this.

mbaldessari committed:
Merge pull request #284 from mbaldessari/wip-next-gen-bandini2

Revert "Drop double manuela-ci namespace"

mbaldessari committed:
Drop old test diffs

mbaldessari committed:
Drop old test diffs

mbaldessari committed:
Merge pull request #285 from mbaldessari/drop-old-tests

Drop old test diffs

mbaldessari committed:
Merge remote-tracking branch 'common/main' into newer-common

mbaldessari committed:
Merge pull request #286 from mbaldessari/newer-common

Newer common

mbaldessari committed:
Enable temperature sensors on the factory

This makes for a better demo

mbaldessari committed:
Enable temperature sensors on the factory

This makes for a better demo

mbaldessari committed:
Fix the topics on factory

Let's use the topic names as set in the value files and not hard-coded
ones. In fact the old names were not being used at all, since the topics
also get created by the mqtt2kafka bridge

Co-Authored-By: Akos Eros <aeros@redhat.com>

mbaldessari committed:
Fix the topics on factory

Let's use the topic names as set in the value files and not hard-coded
ones. In fact the old names were not being used at all, since the topics
also get created by the mqtt2kafka bridge

Co-Authored-By: Akos Eros <aeros@redhat.com>

mbaldessari committed:
Fix the topics used to fetch from the production kafka

When using mirrormaker2 the topics get prefixed with the name of the
kafka cluster where they originate from, so let's add that by default.

Co-Authored-By: Akos Eros <aeros@redhat.com>

mbaldessari committed:
Fix the topics used to fetch from the production kafka

When using mirrormaker2 the topics get prefixed with the name of the
kafka cluster where they originate from, so let's add that by default.

Co-Authored-By: Akos Eros <aeros@redhat.com>

darkdoc committed:
Merge pull request #287 from mbaldessari/wip-next-gen-mirrormaker

Fix kafka-to-s3 mirroring from factory

mamurak committed:
added anomaly-detection-service in data-science-project

mlabonte-rh committed:
Updates for code consolidation

mlabonte-rh committed:
Remove import, reorder tests

mlabonte-rh committed:
Lint

mlabonte-rh committed:
Lint

mbaldessari committed:
Make sure the aws-connection-* secrets are in ml-test and ml-development

mbaldessari committed:
Drop opendatahub application

mamurak committed:
switching RHOAI channel from alpha to fast

mbaldessari committed:
Merge pull request #289 from mbaldessari/quick-fixes

A couple of fixes

mbaldessari committed:
Switch to traits for kafka integrations

The newly released camel-k 2.5.0 dropped support for the old
spec.configuration mount points and we need to switch to the mount
traits (https://camel.apache.org/camel-k/2.5.x/traits/mount.html)

This seems to mean that the keys inside a secret need to be under the
`application.properties` umbrella, for some java reason that is
escaping us. Amend the s3-anomaly secret creation so that everything
works.

Co-Authored-By: Akos Eros <aeros@redhat.com>

mbaldessari committed:
Merge pull request #290 from mbaldessari/quick-fixes

Switch to traits for kafka integrations

mamurak committed:
Merge branch 'validatedpatterns/wip-next-gen' into mamurak/integrate_rhoai_wip

mamurak committed:
Removed previous version of common to convert to subtree from https://github.com/hybrid-cloud-patterns/common.git main

mamurak committed:
Add 'common/' from commit 'cacdd2374565d9066bd0ab3aa1c9fec149a82e68'

git-subtree-dir: common
git-subtree-mainline: 9bb2d9872a923c11f624da0025f8d3d7344e2abb
git-subtree-split: cacdd2374565d9066bd0ab3aa1c9fec149a82e68

mamurak committed:
re-inserting install into makefile

mamurak committed:
overriding Makefile

mlabonte-rh committed:
Merge pull request #288 from validatedpatterns/refactor_pytest

Refactor pytest

mbaldessari committed:
Switch notebook to use ds-pipeline-config-custom

The Jupyter notebook is configured to bind-mount this
`ds-pipeline-config` secret. The problem is that this secret gets only
generate if you log into the RHOAI dashboard, go to data science
projects and choose the `ml-development` namespace.

Since this secret contains things that we already know, let's just
create another identical one. The RHOAI dashboard created one looks like
this:

    $ oc extract -n ml-development secrets/ds-pipeline-config --to=- --keys=odh_dsp.json 2>/dev/null | jq .
    {
      "display_name": "Data Science Pipeline",
      "metadata": {
        "tags": [],
        "display_name": "Data Science Pipeline",
        "engine": "Argo",
        "auth_type": "KUBERNETES_SERVICE_ACCOUNT_TOKEN",
        "api_endpoint": "https://ds-pipeline-dspa-ml-development.apps.mcg-hub.aws.validatedpatterns.io",
        "public_api_endpoint": "https://rhods-dashboard-redhat-ods-applications.apps.mcg-hub.aws.validatedpatterns.io/experiments/ml-development",
        "cos_auth_type": "KUBERNETES_SECRET",
        "cos_secret": "aws-connection-pipeline-bucket",
        "cos_endpoint": "http://s3.openshift-storage.svc.cluster.local",
        "cos_bucket": "pipeline-bucket",
        "cos_username": "..",
        "cos_password": "..",
        "runtime_type": "KUBEFLOW_PIPELINES"
      },
      "schema_name": "kfp"
    }

The one we create here will have this content:

    $ oc extract -n ml-development secrets/ds-pipeline-config-custom --to=- --keys=odh_dsp.json 2>/dev/null | jq .
    {
      "display_name": "Data Science Pipeline",
      "metadata": {
        "tags": [],
        "display_name": "Data Science Pipeline",
        "engine": "Argo",
        "auth_type": "KUBERNETES_SERVICE_ACCOUNT_TOKEN",
        "api_endpoint": "https://ds-pipeline-dspa-ml-development.apps.mcg-hub.aws.validatedpatterns.io",
        "public_api_endpoint": "https://rhods-dashboard-redhat-ods-applications.apps.mcg-hub.aws.validatedpatterns.io/experiments/ml-development/",
        "cos_auth_type": "KUBERNETES_SECRET",
        "cos_secret": "aws-connection-pipeline-bucket",
        "cos_endpoint": "http://s3.openshift-storage.svc.cluster.local",
        "cos_bucket": "pipeline-bucket",
        "cos_username": "..",
        "cos_password": "../TXbWAnPebRI4r1ICKQ/tZn8O6U",
        "runtime_type": "KUBEFLOW_PIPELINES"
      },
      "schema_name": "kfp"
    }

mbaldessari committed:
Drop apps. from urls as localClusterDomain includes it

mbaldessari committed:
Add workbench-trusted-ca-bundle

Create a configmap and label it with the
`config.openshift.io/inject-trusted-cabundle: 'true'` label so we can
use it to reference all the trusted certs in the cluster and inject
those in the notebooks.

mbaldessari committed:
Merge pull request #291 from mbaldessari/quick-fixes

ds-pipeline-config fixes and url fixes

mbaldessari committed:
Merge pull request #292 from mbaldessari/quick-fixes

Add workbench-trusted-ca-bundle

mbaldessari committed:
Use new workbench images

mbaldessari committed:
Merge pull request #293 from mbaldessari/quick-fixes

Use new workbench images

darkdoc committed:
Cleanup ambiguous deploymentconfig key from svc and deployment selectors

mbaldessari committed:
Merge pull request #294 from darkdoc/cleanup_deploymentconfig

Cleanup ambiguous deploymentconfig key from svc and deployment selectors

darkdoc committed:
Move serviceaccount and rolebinding to pipeline chart

Moving these two resources inside the datacenter/pipelines chart we can
remove the namepspace resource that was duplicated (created with
clustergroup chart) and remove the secrets/pipeline-setup chart since its
no longer needed.

mbaldessari committed:
Add an imperative job to create the manuela-dev repo in gitea

This leverages the imperative framework with its default settings. This
means it will run every 10 minutes and is idempotent, meaning that if
the manuela-dev repo does not exist in gitea it will be cloned from
upstream, but if it exists nothing will be done.

Co-Authored-By: Aros Ekos <aeros@redhat.com>

mbaldessari committed:
Merge pull request #295 from mbaldessari/quick-fixes

Create manuela-dev repo in gitea

mbaldessari committed:
Try and use gitea url by default

Also update the tekton image being used while we're at it.

Co-Authored-By: Akos Eros <aeros@redhat.com>

mbaldessari committed:
Merge pull request #296 from darkdoc/refactor_pipeline_setup

Move serviceaccount and rolebinding to pipeline chart

mbaldessari committed:
Drop git secrets from secret template

We use the in-cluster gitea generated ones anyways

mbaldessari committed:
Drop aws secrets from secret template

We use the nooba s3 backend these days

mbaldessari committed:
Drop argocd-env secret and create it automatically

We can now autocreate it via pushsecrets and externalsecrets

mbaldessari committed:
Make sslVerify a parameter (defaulting to false for now)

mbaldessari committed:
Merge pull request #297 from mbaldessari/quick-fixes

Pipeline fixes

mbaldessari committed:
Switch to different manuela-dev branch

mbaldessari committed:
Switch argo to a certified recent image and drop unneeded argocd-version overrides

mbaldessari committed:
Make the lookuppolicy consistent across all imagestreams

mbaldessari committed:
Push all the branches that we have locally

We push all the things to gitea, because sometimes it seems
git is pushing the current working branch back to main instead of the
current branch in gitea. This fixes a bunch of oddities we were
observing.

Co-Authored-By: Akos Eros <aeros@redhat.com>

mbaldessari committed:
Have the branch in the secret for datascience project

mbaldessari committed:
Drop mistakenly pushed bit

mbaldessari committed:
Merge pull request #298 from mbaldessari/quick-fixes

A bunch of fixes

mbaldessari committed:
Stop forcing an importPolicy in the imageStream

It seems to further confuse argo. In any case the healthcheck for
imagestreams is not 100% working in all corner cases. This reduces the
diff a bit, but is not the full solution.

mbaldessari committed:
Add internal CAs to the jupyter notebook

We bind mount a configmap that has been annotated with
`inject-trusted-cabundle: true`, and we bind mount both
`kube-root-ca.crt` and `openshift-service-ca.crt`.
We concatenate those three together in an initcontainer
and then use that as our custom ca cert file.

mbaldessari committed:
Merge pull request #299 from mbaldessari/quick-fixes

Fixed with imagestreams diff and jupiter notebook tls certs

darkdoc committed:
Add inferenceservices

These are the initial inference service that reference an initial model
that gets uploaded to S3 user-bucket via imperative framework.

The jupyter notebooks will commit to these files and point to a newer
model once they run.

darkdoc committed:
Copy initial model imperative ansible job

This uploads an initial small model so that everything works out of the
box from the start. After the jupiter notebooks get run a different
model will be used.

mbaldessari committed:
Use a custom built iot_consumer for now until we switch this to main

mbaldessari committed:
Enable RHOAI in manuela-tst-all

mbaldessari committed:
Merge pull request #300 from mbaldessari/quick-fixes

Fixes

mbaldessari committed:
Drop spurious labels in namespaces

They are not strictly needed and just cause confusion.

mbaldessari committed:
Merge pull request #301 from mbaldessari/quick-fixes

Some more fixes

mbaldessari committed:
Fix whitespaces

mbaldessari committed:
Fix checking out of staging-approval branch

We do a couple of things here:
- if STARTINGBRANCH is set we create a local branch from this
  STARTINGBRANCH, we call it BRANCH and make sure the upstream is set to
  origin/BRANCH. This way we are sure we always use a local branch
  called BRANCH and ignore if an upstream already had BRANCH named like
  that.

- We force push this BRANCH when we push back to gitea

Co-Authored-By: Akos Eros <aeros@redhat.com>

mbaldessari committed:
Try and fix the PR pipeline

Here we switch to using the proper curl command to create a PR in gitea
by default instead of github. Likely we'd need some changes here if also
need to support github for this process.

Co-Authored-By: Akos Eros <aeros@redhat.com>

mbaldessari committed:
Talk to the model server on the hub for now

This is an initial pass at making a stock deployment to just work.
Basically we tell the messaging component on a factory (which represents
a production workload) to talk to the inference server in the
manuela-tst-all namespace on the hub.

This change will need more in-depth discussion in the future and we need
to clear up the exact workflow. We could decide to spin up an inference
server in the factory as well and have it load a model from the hub's
S3, but it might get a bit too onerous for a demo setup like this one.

So for the time being, let's use this approach to at least get the line
dashboard working on the factory, out of the box.

Co-Authored-By: Akos Eros <aeros@redhat.com>

mbaldessari committed:
Merge pull request #302 from mbaldessari/quick-fixes

More fixes

mbaldessari committed:
Preload juniper notebook with manuela-dev git repo

If the folder does not exist, let's clone from the right git repo and
the right branch at the first startup. This improves the user experience
quite a bit and provides for less need to click things around.

mbaldessari committed:
Split off hard-coded secret from configmap to a separate secret

In a second change we will autogenerate that secret via vault

mbaldessari committed:
Add inference server to the factory

We connect to the s3 on the hub via http because we have not been able
yet to confirgure the hub's CA for the inference server.

Co-Authored-By: Akos Eros <aeros@redhat.com>

mbaldessari committed:
Merge pull request #303 from mbaldessari/quick-fixes

Preload juniper notebook with manuela-dev git repo

mbaldessari committed:
Switch to using a generator + pushsecrets and externalsecrets to manage the iotuser password

The iotpassword is the secret protecting the communication between the
sensors and the mqtt broker. Let's generate it once via a generator,
store it in a secret, upload it to vault and fetch it from there.
This way it's an incluster thing only.

mbaldessari committed:
Add ESO on the factory

mbaldessari committed:
SkipDryRunOnMissingResource for external secrets on factory

Makes the deployment fast as we do not have to wait for ESO to fully
complete

mbaldessari committed:
Merge pull request #304 from mbaldessari/quick-fixes

Drop hard coded password

mbaldessari committed:
Add ACM policy to manage dsci on factories

This is needed because the factory needs a way to trust the HUB's CA.

Co-Authored-By: Akos Eros <aeros@redhat.com>

mbaldessari committed:
Use local on-factory inference server

mbaldessari committed:
Merge pull request #305 from mbaldessari/quick-fixes

Inference server on the factory

mamurak committed:
Merge branch 'wip-next-gen' into mamurak/integrate_rhoai_wip

mamurak committed:
added workbench role binding for elyra

mamurak committed:
added workbench role binding for elyra

mbaldessari committed:
Merge pull request #306 from validatedpatterns/wip-next-gen

Merge IE 2.0 work into main

mbaldessari committed:
We can now switch to manuela-dev's main branch by default

mbaldessari committed:
Merge pull request #307 from mbaldessari/manuela-branch

We can now switch to manuela-dev's main branch by default

mbaldessari committed:
Fix path when invoking the qe run_test.sh script

mbaldessari committed:
Merge pull request #613 from mbaldessari/fix-qe-command

Fix path when invoking the qe run_test.sh script

mbaldessari committed:
Switch to the newly built images and drop the old workaround

Co-Authored-By: Akos Eros <aeros@redhat.com>

mbaldessari committed:
Merge pull request #308 from mbaldessari/quick-fixes

Switch to the newly built images and drop the old workaround

mbaldessari committed:
Merge remote-tracking branch 'common-upstream/main' into common-automatic-update

mbaldessari committed:
Merge pull request #309 from mbaldessari/common-automatic-update

Fix path when invoking the qe run_test.sh script

mbaldessari committed:
Fix subscriptions to be tested in CI with IE 2.0

mbaldessari committed:
Do not check for pods in manuela-ci because chances are there won't be any

This would only work if the test is run while a pipeline is currently
running, which makes no sense. So let's skip it as it brings little
value.

Co-Authored-By: Akos Eros <aeros@redhat.com>

mbaldessari committed:
Ignore __pycache__ folder

mbaldessari committed:
Revert "Enable temperature sensors on the factory"

This reverts commit bcf0809017247f89611d791aba4357e42920914b.

QE expects those sensors to be false for testing reasons

mbaldessari committed:
Merge pull request #310 from mbaldessari/quick-fixes

Some more QE fixes

darkdoc committed:
Refactor pipelines and pipeline runs

Add common, generalized pipeline for all iot app (build and seed)
Cleanup some duplications, and not used pipelines, pipelineruns

darkdoc committed:
Use RH blessed image for tkn task

darkdoc committed:
Fix application string in tests

Labels was changed in eaa738ab

mbaldessari committed:
Use RH blessed images

This will also fix the s2i uploads since buildah includes retries by
default now.

mbaldessari committed:
Use RH blessed image for curl as well

darkdoc committed:
Various image, Makefile, pipeline related cleanup

Fix echo command in s2i step. Removed not used pipelineruns,
targets, and variables from Makefile. Removed images folder which is not
used anywhere.

mbaldessari committed:
Ignore ImageStream tags diffs on factory as well

mbaldessari committed:
Fix some qe tests

mbaldessari committed:
Merge pull request #311 from darkdoc/refactor_pipelines

Pipeline refactor, and misc cleanup

mbaldessari committed:
Drop spurious comma

mbaldessari committed:
Merge pull request #312 from mbaldessari/quick-fixes

Some fixes

mbaldessari committed:
Stop using global for s3 bucket aggregation variable

No need to use a global variable and it is confusing to have it in
values-global.yaml

Co-Authored-By: Akos Eros <aeros@redhat.com>

mbaldessari committed:
Use FQDN when pulling the mlserver image

mbaldessari committed:
Merge pull request #313 from mbaldessari/quick-fixes

More cleanups

mbaldessari committed:
Drop the last non-fqdn image

mbaldessari committed:
Drop unused staging namespace and examples

mbaldessari committed:
Merge pull request #314 from mbaldessari/quick-fixes

More fixes and merging a change from Max's branch

mamurak committed:
Merge branch 'validatedpatterns/main' into mamurak/integrate_rhoai_wip

mamurak committed:
added imperative job to upload raw data to s3 bucket
